executor/src/witgen/query_processor.rs

```diff
@@ -110,24 +109,23 @@ where
                 *op,
                 self.evaluate_expression(right, rows)?,
             )),
-            Expression::UnaryOperation(op, expr) => {
-                // Special case to allow `x'` for now on witness columns.
-                // TODO: Check if we actually need this or should replace it by `x(i + 1)`
-                if *op == UnaryOperator::Next {
-                    if let Expression::Reference(Reference::Poly(poly)) = expr.as_ref() {
-                        return self.evaluate_poly(poly, true, rows);
-                    }
-                }
-                Ok(evaluate_unary_operation(
-                    *op,
-                    self.evaluate_expression(expr, rows)?,
-                ))
-            }
+            Expression::UnaryOperation(op, inner) => Ok(evaluate_unary_operation(
+                *op,
+                self.evaluate_expression(inner, rows)?,
+            )),
             Expression::Reference(Reference::LocalVar(i, _name)) => {
                 assert!(*i == 0);
                 Ok(rows.current_row_index.into())
             }
-            Expression::Reference(Reference::Poly(poly)) => self.evaluate_poly(poly, false, rows),
+            Expression::FunctionCall(FunctionCall { id, arguments }) => {
+                let [arg] = &arguments[..] else {
+                    return Err(IncompleteCause::ExpressionEvaluationUnimplemented(
```

Comment:
In general, shouldn't we remove `IncompleteCause::ExpressionEvaluationUnimplemented` and instead panic in such cases? If we return incomplete, the query will effectively be ignored without any error message, and it would be quite hard for the user to find out why witgen didn't succeed.

Reply:
Yes, I would suggest to revisit the error cases when the evaulator is complete. This code here will be removed in another PR, but the evaluator currently creates a lot of String errors which might not be necessary and we need to check if we always panic in such cases.